### PR TITLE
Sidebar: Site Picker

### PR DESCRIPTION
**Pull request by kean**
Integrate Site Picker and Site Creation flows in the new sidebar.

https://github.com/user-attachments/assets/b3327899-008f-4b62-a2fa-07a943c37e97

I also updated the new self-hosted site flow to be presented modally as other similar flows.

Screenshot 2024-08-26 at 6 05 27 PM

## Known Issues

The sidebar/split view needs to react to the WordPressAuthentificator notifications to reload the sites after creation. It currently doesn't  automatically.

## To test:

- Login with an account that has 4 or more sites
- Verify that you can add sites via the "All Sites" screen and via the "Add Site" button
- Verify that you can change sites using the "All Sites" screen
- (Regressing Testing) Verify the same steps for the existing site picker in the Home page

## Regression Notes
1. Potential unintended areas of impact

2. What I did to test those areas of impact (or what existing automated tests I relied on)

3. What automated tests I added (or what prevented me from doing so)

PR submission checklist:

- [ ] I have completed the Regression Notes.
- [ ] I have considered adding unit tests for my changes.
- [ ] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">PR requires at least one label.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Modules/Sources/DesignSystem/Components/DSButton.swift [hunk 1]
```diff
@@ -147,12 +147,12 @@ public struct DSButton: View {
         return disabledOpacity
     }
 
-    private var priamryDisabledOpacity: CGFloat {
+    private var primaryDisabledOpacity: CGFloat {
         isEnabled ? 1 : disabledOpacity
     }
 
     private var disabledOpacity: CGFloat {
-        colorScheme == .light ? 0.5 : 0.6
```
Comment (kean) [inline Modules/Sources/DesignSystem/Components/DSButton.swift:155]: The dark gray looked bad.

File: WordPress/Classes/Login/LoginWithUrlView.swift [hunk 2]
```diff
@@ -33,14 +33,14 @@ struct LoginWithUrlView: View {
...
         VStack(alignment: .leading) {
-            VStack(alignment: .center, spacing: 16) {
-                Image("splashLogo")
-                    .resizable()
-                    .aspectRatio(contentMode: .fit)
-                    .frame(width: 60, height: 60)
-                    .padding(24)
-                Text(Self.enterSiteAddress)
-            }
+            Image("splashLogo")
```
Comment (kean) [inline WordPress/Classes/Login/LoginWithUrlView.swift:36]: Fixes horizontal alignment on larger screens.

File: WordPress/Classes/System/SplitViewRootPresenter.swift [hunk 3]
```diff
@@ -110,6 +114,28 @@ final class SplitViewRootPresenter: RootViewPresenter {
...
     }
 
+    private func showSitePicker() {
+        let sitePickerVC = SiteSwitcherViewController(
+            addSiteAction: { [weak self] in
+                self?.showAddSiteScreen(selection: $0)
+            },
+            onSiteSelected: { [weak self] site in
+                self?.splitVC.dismiss(animated: true)
+                RecentSitesService().touch(blog: site)
```
Comment (kean) [inline WordPress/Classes/System/SplitViewRootPresenter.swift:125]: This call happens all over the codebase, so I copy-pasted it too for now.

File: WordPress/Classes/ViewRelated/System/Sidebar/SiteMenuViewController.swift [hunk 4]
```diff
@@ -43,7 +43,7 @@ final class SiteMenuViewController: UIViewController {
 private final class SiteMenuListViewController: BlogDetailsViewController {
     override func tableView(_ tableView: UITableView, heightForHeaderInSection section: Int) -> CGFloat {
         let title = super.tableView(tableView, titleForHeaderInSection: section)
-        return title == nil ? (section == 0 ? 0 : 20) : 52
+        return title == nil ? 0 : 48
```
Comment (kean) [inline WordPress/Classes/ViewRelated/System/Sidebar/SiteMenuViewController.swift:46]: Removing ghost sections in the new site menu. The app no longer uses the grouped style, so these extra vertical spacing made no sense. In the new menu, all items now fit vertically, even "Site Settings".

**Comment by kean** [comment]
I've made a couple of changes since I initially opened this PR:

- Update the design of sites in the sidebar based on the feedback – the new version is the best of both worlds
- Fix an issue with blog not being selected after a new site is added (both self-hosted and Dotcom)
- Update "Add Site" icon design to match what Apple usually does and make it look different from the regular menu items
- Fix selection color in site menu
- Hide recent sites from All Sites view

sidebar-3

File: WordPress/Classes/ViewRelated/Blog/Site Picker/BlogList/SiteIconViewModel.swift [hunk 5]
```diff
@@ -15,7 +15,7 @@ struct SiteIconViewModel {
 
         var width: CGFloat {
             switch self {
-            case .small: 24
+            case .small: 28
```
Comment (kean) [inline WordPress/Classes/ViewRelated/Blog/Site Picker/BlogList/SiteIconViewModel.swift:18]: I tested other places where it is used – nav bar in posts, etc – and it looks good with a new size.

**Review by crazytonyli — APPROVED** [review]

**Pull request merged**